commcare_connect/opportunity/tests/test_management_commands.py

```
@@ -42,3 +57,190 @@ def test_archive_pending_invoices(self):
+
+        visit.refresh_from_db()
+        completed_work.refresh_from_db()
+        assert visit.status == VisitValidationStatus.rejected
+        assert visit.reason == "Manual ops fix per CCCT-XXXX"
+        assert completed_work.status == CompletedWorkStatus.rejected
+        assert completed_work.saved_payment_accrued == 0
+
+    def test_refuses_when_opportunity_not_found(self, approved_visit_with_completed_work):
+        visit = approved_visit_with_completed_work
+        bogus_opp = type("X", (), {"opportunity_id": "00000000-0000-0000-0000-000000000000"})
+        with pytest.raises(CommandError, match="Opportunity .* not found"):
```

Ruff (0.15.12)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)